diff --git a/include/linux.h b/include/linux.h
index cea468d2b9..3ea9016272 100644
--- a/include/linux.h
+++ b/include/linux.h
@@ -214,7 +214,7 @@
  * fsxattr
  */
 struct file_attr {
-	__u32	fa_xflags;
+	__u64	fa_xflags;
 	__u32	fa_extsize;
 	__u32	fa_nextents;
 	__u32	fa_projid;
diff --git a/libfrog/file_attr.c b/libfrog/file_attr.c
index c2cbcb4e14..6801c54588 100644
--- a/libfrog/file_attr.c
+++ b/libfrog/file_attr.c
@@ -114,7 +114,7 @@
 
 	file_attr_to_fsxattr(fa, &fsxa);
 
-	error = ioctl(fd, FS_IOC_FSSETXATTR, fa);
+	error = ioctl(fd, FS_IOC_FSSETXATTR, &fsxa);
 	close(fd);
 
 	return error;


