diff -urNp -x '*.orig' xfsprogs-5.11.0.org/libxfs/Makefile xfsprogs-5.11.0/libxfs/Makefile
--- xfsprogs-5.11.0.org/libxfs/Makefile	2020-12-23 02:12:40.000000000 +0100
+++ xfsprogs-5.11.0/libxfs/Makefile	2021-03-14 00:46:57.929250648 +0100
@@ -110,7 +110,7 @@ CFILES = cache.c \
 
 FCFLAGS = -I.
 
-LTLIBS = $(LIBPTHREAD) $(LIBRT)
+LTLIBS = $(LIBPTHREAD) $(LIBRT) $(LIBUUID)
 
 # don't try linking xfs_repair with a debug libxfs.
 DEBUG = -DNDEBUG
diff -urNp -x '*.orig' xfsprogs-5.11.0.org/m4/manual_format.m4 xfsprogs-5.11.0/m4/manual_format.m4
--- xfsprogs-5.11.0.org/m4/manual_format.m4	2020-12-23 02:07:48.000000000 +0100
+++ xfsprogs-5.11.0/m4/manual_format.m4	2021-03-14 00:46:57.929250648 +0100
@@ -6,12 +6,5 @@
 #
 AC_DEFUN([AC_MANUAL_FORMAT],
   [ have_zipped_manpages=false
-    for d in ${prefix}/share/man ${prefix}/man ; do
-        if test -f $d/man1/man.1.gz
-        then
-            have_zipped_manpages=true
-            break
-        fi
-    done
     AC_SUBST(have_zipped_manpages)
   ])
