diff -ur xfsprogs-3.0.0.org/libxfs/Makefile xfsprogs-3.0.0/libxfs/Makefile
--- xfsprogs-3.0.0.org/libxfs/Makefile	2009-01-28 20:13:16.000000000 +0100
+++ xfsprogs-3.0.0/libxfs/Makefile	2009-02-04 23:09:19.331922399 +0100
@@ -41,7 +41,9 @@
 include $(BUILDRULES)
 
 install: default
+	$(INSTALL_LTLIB)
 
 install-dev: default
+	$(INSTALL_LTLIB_DEV)
 
 install-qa: default
diff -ur xfsprogs-3.0.0.org/libxlog/Makefile xfsprogs-3.0.0/libxlog/Makefile
--- xfsprogs-3.0.0.org/libxlog/Makefile	2009-01-28 20:13:17.000000000 +0100
+++ xfsprogs-3.0.0/libxlog/Makefile	2009-02-04 23:09:37.878049126 +0100
@@ -12,6 +12,8 @@
 
 CFILES = xfs_log_recover.c util.c
 
+LTLIBS = $(LIBUUID) $(LIBXFS)
+
 # don't want to link xfs_repair with a debug libxlog.
 DEBUG = -DNDEBUG
 
@@ -21,6 +21,12 @@
 
 include $(BUILDRULES)
 
-install install-dev install-qa: default
+install: default                                                                                                                                            
+	$(INSTALL_LTLIB)                                                                                                                                     
+
+install-dev: default                                                                                                                                        
+	$(INSTALL_LTLIB_DEV)                                                                                                                                 
+
+install-qa: default
 
 -include .dep
