--- xfsprogs-2.6.25/include/builddefs.in.orig	2004-10-13 02:26:03.000000000 +0200
+++ xfsprogs-2.6.25/include/builddefs.in	2004-10-17 16:10:37.832771624 +0200
@@ -113,7 +113,7 @@
 PCFLAGS = -I/usr/local/include
 endif
 
-GCFLAGS = -O1 $(OPTIMIZER) $(DEBUG) -funsigned-char -fno-strict-aliasing -Wall \
+GCFLAGS = $(OPTIMIZER) $(DEBUG) -funsigned-char -fno-strict-aliasing -Wall \
 	  -DVERSION=\"$(PKG_VERSION)\" -DLOCALEDIR=\"$(PKG_LOCALE_DIR)\"  \
 	  -DPACKAGE=\"$(PKG_NAME)\" -I$(TOPDIR)/include
 
