--- xfsprogs-2.5.4/m4/manual_format.m4.orig	2003-07-25 19:42:03.000000000 +0200
+++ xfsprogs-2.5.4/m4/manual_format.m4	2003-08-08 08:22:59.000000000 +0200
@@ -6,12 +6,5 @@
 # 
 AC_DEFUN([AC_MANUAL_FORMAT],
   [ have_zipped_manpages=false
-    for d in ${prefix}/share/man ${prefix}/man ; do
-        if test -f $d/man1/man.1.gz
-        then
-            have_zipped_manpages=true
-            break
-        fi
-    done
     AC_SUBST(have_zipped_manpages)
   ])

