diff -Nur xfsprogs-2.8.11.orig/include/builddefs.in xfsprogs-2.8.11.LDFLAGS/include/builddefs.in
--- xfsprogs-2.8.11.orig/include/builddefs.in	2006-07-17 01:22:02.000000000 +0200
+++ xfsprogs-2.8.11.LDFLAGS/include/builddefs.in	2006-08-24 13:13:29.000000000 +0200
@@ -24,6 +24,7 @@
 OPTIMIZER = @opt_build@
 MALLOCLIB = @malloc_lib@
 LOADERFLAGS = @LDFLAGS@
+LTLDFLAGS = @LDFLAGS@
 
 LIBRT = @librt@
 LIBUUID = @libuuid@
diff -Nur xfsprogs-2.8.11.orig/libxfs/Makefile xfsprogs-2.8.11.LDFLAGS/libxfs/Makefile
--- xfsprogs-2.8.11.orig/libxfs/Makefile	2006-08-24 13:13:28.000000000 +0200
+++ xfsprogs-2.8.11.LDFLAGS/libxfs/Makefile	2006-08-24 13:13:57.000000000 +0200
@@ -34,6 +34,8 @@
 
 FCFLAGS = -I.
 
+LTLIBS = $(LIBPTHREAD)
+
 # don't try linking xfs_repair with a debug libxfs.
 DEBUG = -DNDEBUG
 
