diff -uNr xfsprogs-2.6.25/db/agf.h xfsprogs-2.6.25.fixed/db/agf.h
--- xfsprogs-2.6.25/db/agf.h	2004-10-13 02:26:02.000000000 +0200
+++ xfsprogs-2.6.25.fixed/db/agf.h	2005-05-11 03:32:57.000000000 +0200
@@ -30,7 +30,7 @@
  * http://oss.sgi.com/projects/GenInfo/SGIGPLNoticeExplan/
  */
 
-struct field;
+#include "field.h"
 
 extern const struct field	agf_flds[];
 extern const struct field	agf_hfld[];
diff -uNr xfsprogs-2.6.25/db/agfl.h xfsprogs-2.6.25.fixed/db/agfl.h
--- xfsprogs-2.6.25/db/agfl.h	2004-10-13 02:26:02.000000000 +0200
+++ xfsprogs-2.6.25.fixed/db/agfl.h	2005-05-11 03:33:09.000000000 +0200
@@ -30,7 +30,7 @@
  * http://oss.sgi.com/projects/GenInfo/SGIGPLNoticeExplan/
  */
 
-struct field;
+#include "field.h"
 
 extern const struct field	agfl_flds[];
 extern const struct field	agfl_hfld[];
diff -uNr xfsprogs-2.6.25/db/agi.h xfsprogs-2.6.25.fixed/db/agi.h
--- xfsprogs-2.6.25/db/agi.h	2004-10-13 02:26:02.000000000 +0200
+++ xfsprogs-2.6.25.fixed/db/agi.h	2005-05-11 03:33:23.000000000 +0200
@@ -30,7 +30,7 @@
  * http://oss.sgi.com/projects/GenInfo/SGIGPLNoticeExplan/
  */
 
-struct field;
+#include "field.h"
 
 extern const struct field	agi_flds[];
 extern const struct field	agi_hfld[];
diff -uNr xfsprogs-2.6.25/db/dquot.h xfsprogs-2.6.25.fixed/db/dquot.h
--- xfsprogs-2.6.25/db/dquot.h	2004-10-13 02:26:02.000000000 +0200
+++ xfsprogs-2.6.25.fixed/db/dquot.h	2005-05-11 03:32:40.000000000 +0200
@@ -30,7 +30,7 @@
  * http://oss.sgi.com/projects/GenInfo/SGIGPLNoticeExplan/
  */
 
-struct field;
+#include "field.h"
 
 extern const struct field	disk_dquot_flds[];
 extern const struct field	dqblk_flds[];
diff -uNr xfsprogs-2.6.25/db/faddr.h xfsprogs-2.6.25.fixed/db/faddr.h
--- xfsprogs-2.6.25/db/faddr.h	2004-10-13 02:26:02.000000000 +0200
+++ xfsprogs-2.6.25.fixed/db/faddr.h	2005-05-11 03:37:39.000000000 +0200
@@ -30,6 +30,9 @@
  * http://oss.sgi.com/projects/GenInfo/SGIGPLNoticeExplan/
  */
 
+#ifndef __faddr_h__
+#define __faddr_h__
+
 typedef void (*adfnc_t)(void *obj, int bit, typnm_t next);
 
 extern void	fa_agblock(void *obj, int bit, typnm_t next);
@@ -48,3 +51,5 @@
 extern void	fa_dfiloffa(void *obj, int bit, typnm_t next);
 extern void	fa_ino4(void *obj, int bit, typnm_t next);
 extern void	fa_ino8(void *obj, int bit, typnm_t next);
+
+#endif
diff -uNr xfsprogs-2.6.25/db/field.h xfsprogs-2.6.25.fixed/db/field.h
--- xfsprogs-2.6.25/db/field.h	2004-10-13 02:26:02.000000000 +0200
+++ xfsprogs-2.6.25.fixed/db/field.h	2005-05-11 03:37:52.000000000 +0200
@@ -30,6 +30,13 @@
  * http://oss.sgi.com/projects/GenInfo/SGIGPLNoticeExplan/
  */
 
+#ifndef __field_h__
+#define __field_h__
+
+#include "type.h"
+#include "faddr.h"
+#include "fprint.h"
+
 typedef enum fldt	{
 	FLDT_AEXTNUM,
 	FLDT_AGBLOCK,
@@ -222,3 +229,5 @@
 				  void *obj, int startoff);
 extern int		fsize(const field_t *f, void *obj, int startoff,
 				  int idx);
+
+#endif
diff -uNr xfsprogs-2.6.25/db/fprint.h xfsprogs-2.6.25.fixed/db/fprint.h
--- xfsprogs-2.6.25/db/fprint.h	2004-10-13 02:26:02.000000000 +0200
+++ xfsprogs-2.6.25.fixed/db/fprint.h	2005-05-11 03:36:43.000000000 +0200
@@ -30,6 +30,9 @@
  * http://oss.sgi.com/projects/GenInfo/SGIGPLNoticeExplan/
  */
 
+#ifndef __fprint_h__
+#define __fprint_h__
+
 typedef int (*prfnc_t)(void *obj, int bit, int count, char *fmtstr, int size,
 		       int arg, int base, int array);
 
@@ -43,3 +46,5 @@
 			int arg, int base, int array);
 extern int	fp_uuid(void *obj, int bit, int count, char *fmtstr, int size,
 			int arg, int base, int array);
+
+#endif
diff -uNr xfsprogs-2.6.25/db/inode.h xfsprogs-2.6.25.fixed/db/inode.h
--- xfsprogs-2.6.25/db/inode.h	2004-10-13 02:26:02.000000000 +0200
+++ xfsprogs-2.6.25.fixed/db/inode.h	2005-05-11 03:33:57.000000000 +0200
@@ -30,6 +30,8 @@
  * http://oss.sgi.com/projects/GenInfo/SGIGPLNoticeExplan/
  */
 
+#include "field.h"
+
 extern const struct field	inode_a_flds[];
 extern const struct field	inode_core_flds[];
 extern const struct field	inode_flds[];
diff -uNr xfsprogs-2.6.25/db/sb.h xfsprogs-2.6.25.fixed/db/sb.h
--- xfsprogs-2.6.25/db/sb.h	2004-10-13 02:26:03.000000000 +0200
+++ xfsprogs-2.6.25.fixed/db/sb.h	2005-05-11 03:34:26.000000000 +0200
@@ -30,7 +30,7 @@
  * http://oss.sgi.com/projects/GenInfo/SGIGPLNoticeExplan/
  */
 
-struct field;
+#include "field.h"
 
 extern const struct field	sb_flds[];
 extern const struct field	sb_hfld[];
diff -uNr xfsprogs-2.6.25/db/type.h xfsprogs-2.6.25.fixed/db/type.h
--- xfsprogs-2.6.25/db/type.h	2004-10-13 02:26:03.000000000 +0200
+++ xfsprogs-2.6.25.fixed/db/type.h	2005-05-11 03:34:57.000000000 +0200
@@ -30,6 +30,9 @@
  * http://oss.sgi.com/projects/GenInfo/SGIGPLNoticeExplan/
  */
 
+#ifndef __type_h__
+#define __type_h__
+
 struct field;
 
 #define	szof(x,y)	sizeof(((x *)0)->y)
@@ -68,3 +71,5 @@
 			      char **argv);
 extern void	handle_text(int action, const struct field *fields, int argc,
 			    char **argv);
+
+#endif
diff -uNr xfsprogs-2.6.25/include/swab.h xfsprogs-2.6.25.fixed/include/swab.h
--- xfsprogs-2.6.25/include/swab.h	2004-10-13 02:26:03.000000000 +0200
+++ xfsprogs-2.6.25.fixed/include/swab.h	2005-05-11 03:38:39.000000000 +0200
@@ -108,7 +108,7 @@
  __fswab64((x)))
 
 
-static __inline__ __const__ __u16 __fswab16(__u16 x)
+static __inline__ __u16 __fswab16(__u16 x)
 {
 	return __arch__swab16(x);
 }
@@ -121,7 +121,7 @@
 	__arch__swab16s(addr);
 }
 
-static __inline__ __const__ __u32 __fswab32(__u32 x)
+static __inline__ __u32 __fswab32(__u32 x)
 {
 	return __arch__swab32(x);
 }
@@ -134,7 +134,7 @@
 	__arch__swab32s(addr);
 }
 
-static __inline__ __const__ __u64 __fswab64(__u64 x)
+static __inline__ __u64 __fswab64(__u64 x)
 {
 #  ifdef __SWAB_64_THRU_32__
 	__u32 h = x >> 32;
diff -uNr xfsprogs-2.6.25/mkfile/xfs_mkfile.c xfsprogs-2.6.25.fixed/mkfile/xfs_mkfile.c
--- xfsprogs-2.6.25/mkfile/xfs_mkfile.c	2004-10-13 02:26:06.000000000 +0200
+++ xfsprogs-2.6.25.fixed/mkfile/xfs_mkfile.c	2005-05-11 03:41:57.000000000 +0200
@@ -39,7 +39,7 @@
 
 #define	MAXBUFFERSIZE	(256 * 1024)
 
-static char *progname;
+char *progname;
 
 static void
 usage(void)
