diff -urN ./old/xfsprogs-1.3.5/configure.in xfsprogs-1.3.5/configure.in
--- ./old/xfsprogs-1.3.5/configure.in	Thu Aug  9 21:04:48 2001
+++ xfsprogs-1.3.5/configure.in	Wed Aug 29 09:12:23 2001
@@ -201,14 +201,6 @@
 dnl (usually on Debian, but not Redhat pre-7.0)
 pkg_man_dir=${prefix}/share/man
 have_zipped_manpages=false
-for d in ${prefix}/share/man ${prefix}/man ; do
-    if test -f $d/man1/man.1.gz
-    then
-	pkg_man_dir=$d
-	have_zipped_manpages=true
-	break
-    fi
-done
 AC_SUBST(pkg_man_dir)
 AC_SUBST(have_zipped_manpages)
 
